### PR TITLE
Phase 2 tickets

### PR DESCRIPTION
**Pull request by yeeaaagz**
ENG-256170: BarChart: Skin chart to new design
ENG-256172: BarChart: Add API Documentation and Examples
ENG-256174: Bar Chart: Add No Data and Corrupted Data State
ENG-256177: BarChart: Add Brushing
ENG-256180: BarChart: Add Legend
ENG-256184: BarChart: Gridline support

### CODE REVIEW

File: src/ntnx/cartesian/CartesianAxis.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    y: 0,
+    width: 0,
+    height: 0,
+    viewBox: { x: 0, y: 0, width: 0, height: 0 },
+    // The orientation of axis
+    orientation: 'bottom',
+    // The ticks
+    ticks: [],
+
+    stroke: '#666',
```
Comment (shanechu-ntnx) [inline src/ntnx/cartesian/CartesianAxis.js]: static color?

File: src/ntnx/cartesian/Brush.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+     * Leave time out.
+     */
+    leaveTimeOut: PropTypes.number,
+  };
+
+  static defaultProps = {
+    height: 15,
+    travellerWidth: 5,
+    gap: 1,
+    fill: '#fff',
```
Comment (shanechu-ntnx) [inline src/ntnx/cartesian/Brush.js]: static color?

File: src/ntnx/cartesian/Bar.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    /**
+     * The customized event handler of mouseover on the bars in this group.
+     */
+    onMouseOver: PropTypes.func,
+    /**
+     * The customized event handler of mouseout on the bars in this group.
+     */
+    onMouseOut: PropTypes.func,
+    /**
+     * The customized event handler of moustenter on the bars in this group.
```
Comment (shanechu-ntnx) [inline src/ntnx/cartesian/Bar.js]: typo (moustenter)

File: src/ntnx/cartesian/Bar.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    data: PropTypes.arrayOf(PropTypes.shape({
+      x: PropTypes.number,
+      y: PropTypes.number,
+      width: PropTypes.number,
+      height: PropTypes.number,
+      radius: PropTypes.oneOfType([PropTypes.number, PropTypes.array]),
+      value: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.array]),
+    })),
+    /**
+     * The customized event handler of animtion start.
```
Comment (shanechu-ntnx) [inline src/ntnx/cartesian/Bar.js]: animation typo

File: src/ntnx/cartesian/Bar.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+     */
+    onAnimationStart: PropTypes.func,
+    /**
+     * The customized event handler of animtion end.
```
Comment (shanechu-ntnx) [inline src/ntnx/cartesian/Bar.js]: same here

File: src/ntnx/cartesian/Brush.js [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        <rect
+          className="recharts-brush-slide-grabber ntnx"
+          onMouseEnter={this.handleEnterSlideOrTraveller}
+          onMouseLeave={this.handleLeaveSlideOrTraveller}
+          onMouseDown={this.handleSlideDragStart}
+          onTouchStart={this.handleSlideDragStart}
+          style={{ cursor: 'move' }}
+          fill='rgba(255, 255, 255, 0)'
+          x={Math.min(startX, endX)}
+          y={y}
```
Comment (shanechu-ntnx) [inline src/ntnx/cartesian/Brush.js:137]: Is linting on for this project? No spaces in between brackets here, but there are spaces on the next line. Also noticed some other linting inconsistencies in various places

File: src/ntnx/component/Tooltip.js [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+import classNames from 'classnames';
+import DefaultTooltipContent from '../../component/DefaultTooltipContent';
+import { TOOLTIP_TYPES, isSsr } from '../../util/ReactUtils';
+import { isNumber } from '../../util/DataUtils';
+import pureRender from '../../util/PureRender';
+import ReTooltip from '../../component/Tooltip';
+
+const CLS_PREFIX = 'recharts-tooltip-wrapper';
+
+const EPS = 1;
```
Comment (shanechu-ntnx) [inline src/ntnx/component/Tooltip.js]: What's this? Didn't notice this being used

**Review by shanechu-ntnx — APPROVED** [review]

File: src/ntnx/cartesian/Bar.js (hunk 3 above)
Comment (yeeaaagz) [reply src/ntnx/cartesian/Bar.js]: done thx

File: src/ntnx/cartesian/Bar.js (hunk 4 above)
Comment (yeeaaagz) [reply src/ntnx/cartesian/Bar.js]: done

File: src/ntnx/cartesian/Bar.js (hunk 5 above)
Comment (yeeaaagz) [reply src/ntnx/cartesian/Bar.js]: done

File: src/ntnx/cartesian/Brush.js (hunk 2 above)
Comment (yeeaaagz) [reply src/ntnx/cartesian/Brush.js]: done

File: src/ntnx/cartesian/Brush.js (hunk 6 above)
Comment (yeeaaagz) [reply src/ntnx/cartesian/Brush.js:137]: No linting installed. When We move this to the nutanix-core. I will install linting and fix all the inconsistencies.

File: src/ntnx/cartesian/CartesianAxis.js (hunk 1 above)
Comment (yeeaaagz) [reply src/ntnx/cartesian/CartesianAxis.js]: done

File: src/ntnx/component/Tooltip.js (hunk 7 above)
Comment (yeeaaagz) [reply src/ntnx/component/Tooltip.js]: done

**Pull request merged**